# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/ipc/PWebGPU.ipdl
# Commit: 41d7ed93df76
# Full Hash: 41d7ed93df76d84a8e842758f852d291a908e065
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2023-10-17 03:58:35
# Regressor Bug: 1838693
# File Overlap Count: 2
# Description:
#   Bug 1838693 Part 3: Make GPUDevice::Destroy() trigger wgpu device_destroy, then device_drop. r=webgpu-reviewers,nical
#   
#   This treats destroy as a 2-step process: wait on the destroy, then drop.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D190238
# ==============================================================================

diff -r f214290956c1 -r 41d7ed93df76 dom/webgpu/ipc/PWebGPU.ipdl
--- a/dom/webgpu/ipc/PWebGPU.ipdl	Mon Oct 16 15:32:00 2023 +0000
+++ b/dom/webgpu/ipc/PWebGPU.ipdl	Mon Oct 16 15:32:00 2023 +0000
@@ -59,6 +59,7 @@
   async TextureViewDestroy(RawId selfId);
   async SamplerDestroy(RawId selfId);
   async DeviceDestroy(RawId selfId);
+  async DeviceDrop(RawId selfId);
 
   async CommandEncoderFinish(RawId selfId, RawId deviceId, GPUCommandBufferDescriptor desc);
   async CommandEncoderDestroy(RawId selfId);